internal/automation-security@-emails/automationSecurity@Emails.gs

```diff
@@ -0,0 +1,324 @@
+      },
+      "reporter": {
+        "name": reporter
+      },
+      "description": "Reporter: "+ from +" \nDate: "+ Utilities.formatDate(new Date(), "IST", "yyyy-MM-dd'/'HH:mm:ss") + "\nReference: Please find the \""+ subject +"\" in Security@." 
+    }
+  };
+  
+  var url = "https://WSO2_JIRA_DOMAIN/jira/rest/api/2/issue/";
+  var payload = JSON.stringify(bodyData);
+
+  var headers = { "Accept":"application/json", 
+              "Content-Type":"application/json", 
+              "Authorization":"Basic " + encCred,
+         };
+
+  var options = { "method":"POST",
+              "contentType" : "application/json",
+              "headers": headers,
+              "payload" : payload
+           };
+  
+ var response = UrlFetchApp.fetch(url, options);
+ 
+  if(response.getResponseCode() == 201){
+    
+    var res = response.getContentText();
+    var JSONres = JSON.parse(res);
+    
+    var key = JSONres["key"]
+
+    if(key != null){
+      var iurl = "https://WSO2_JIRA_DOMAIN/jira/browse/"+key;
+    }
+    
+    var returnMessage = sendMail(cc, subject, iurl, threadID); 
+    
+    return {
+        ticketMsg: "Jira ticket successfully created!", 
+        mailMsg: returnMessage
+    };
+  }
+  else{
+    
+    var err = JSON.parse(response.getContentText());
+    
+    return {
+      ticketMsg: "Error when creating the ticket!"+err, 
+      mailMsg: "Could not send the reply due to ticket creation failure!"
+    };
+  
+  }
+
+}
+
+/*
+* sendMail() -- After creating the ticket sends nd auto reply email to the same thread along with the ticket URL
+*
+*@param CC Array containing any CC participants of the email
+*@param subject Subject of the email
+*@param issueURL URL of the ticket created 
+*@param threadId Thread ID of the email
+*
+*@return returns the success message of the auto-reply email
+*
+*/
+
+function sendMail(CC, subject, issueUrl, threadId){ 
+  
+  var sender = userId; // (from)
+  var reciever = reciever1; // (to)
+  var cc = CC.toString(); 
+  var username = userId 
+  
+  var message = "From: "+ sender +"\nTo: "+ reciever +"\nSubject: "+ subject +"\nCc: "+ cc +"\nDate: "+ Utilities.formatDate(new Date(), "GMT+1", "dd/MM/yyyy") +" \nMessage-ID: sentFromGAPI \n\nA JIRA issue has been created with the following URL - "+ issueUrl;
+  var encodedmessage = Utilities.base64EncodeWebSafe(message);
+  
+  var resource = {
+    "raw": encodedmessage,
+    "payload": {
+      "headers": [
+        {
+          "name": "to",
+          "value": reciever
+        },
+        {
+          "name": "from",
+          "value": sender
+        },
+        {
+          "name": "subject",
+          "value": subject
+        },
+        {
+          "name": "cc",
+          "value": cc
+        }
+      ],
+      "mimeType": "text/plain"
+    },
+   "threadId": threadId
+  }
+  
+  var payload = JSON.stringify(resource);
+
+  var Requesturl = "https://www.googleapis.com/gmail/v1/users/"+username+"/messages/send";
```

Comment:
Better to move API URLs to the config file.

Reply:
Done!